server/src/main/java/org/elasticsearch/index/shard/GlobalCheckpointListeners.java

```diff
@@ -193,22 +190,31 @@ synchronized void globalCheckpointUpdated(final long globalCheckpoint) {
     private void notifyListeners(final long globalCheckpoint, final IndexShardClosedException e) {
         assert Thread.holdsLock(this);
         assert (globalCheckpoint == UNASSIGNED_SEQ_NO && e != null) || (globalCheckpoint >= NO_OPS_PERFORMED && e == null);
```

Comment:
Should we break this assertion into two branches of the following if-else statement?

Reply:
Sure thing, I will do this in a follow-up though as I want to unblock the main effort, and make a small adjustment to another assertion in this file.

Reply:
@dnhatn See #33722.